dkg/sync/client.go

```diff
@@ -63,9 +63,7 @@ func (c *Client) Run(ctx context.Context) error {
 			return ctx.Err()
 		}
 
-		retry := c.reconnect // Retry connecting if never connected.
```

Comment:
the issue is that we copied this variable and passed it into `connect` below which blocked forever retrying even though `DisableReconnect` was called a little after this.